src/entity/object/ExperienceOrb.php

```diff
@@ -145,15 +145,19 @@ public function getTargetPlayer() : ?Human{
 		}
 
 		$entity = $this->getWorld()->getEntity($this->targetPlayerRuntimeId);
-		if($entity instanceof Human){
+		if($entity instanceof Human && !$entity->getXpManager()->isAttractsLocked()){
 			return $entity;
 		}
 
 		return null;
 	}
 
 	public function setTargetPlayer(?Human $player) : void{
-		$this->targetPlayerRuntimeId = $player !== null ? $player->getId() : null;
+		if(($player instanceof Human && !$player->getXpManager()->isAttractsLocked()) or $player === null){
+			$this->targetPlayerRuntimeId = $player !== null ? $player->getId() : null;
+		}else{
+			$this->targetPlayerRuntimeId = null;
+		}
```

Comment:
This looks really overcomplicated. This would be much simpler
```php
if($player !== null && !$player->getXpManager()->isAttractsLocked()){
			$this->targetPlayerRuntimeId = $player->getId();
		}else{
			$this->targetPlayerRuntimeId = null;
		}
```
or
```php
$this->targetPlayerRuntimeId = $player !== null && !$player->getXpManager()->isAttractsLocked() ? $player->getId() : null;
```

But does this check even belong there? In every case, where pm would set the target player itself, it does the checks alone. So shouldn't plugin devs be required to do their own checks? Possibly someone wants to bypass the lock of another plugin (whyever).

Reply:
No, it doesn't belong here. If the player can't be tracked, the orb should try to find a different target.

Reply:
This whole change is redundant.